clang/test/CIR/CodeGen/builtin-prefetch.c

```diff
@@ -0,0 +1,20 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -emit-cir %s -o - | FileCheck %s -check-prefix=CIR
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-llvm %s -o - | FileCheck %s -check-prefix=LLVM
+
+void foo(void *a) {
+  __builtin_prefetch(a, 1, 1);
+}
+
+// CIR:  cir.func @foo(%arg0: !cir.ptr<!void> loc({{.*}}))
+// CIR:    [[PTR_ALLOC:%.*]] = cir.alloca !cir.ptr<!void>, cir.ptr <!cir.ptr<!void>>, ["a", init] {alignment = 8 : i64}
+// CIR:    cir.store %arg0, [[PTR_ALLOC]] : !cir.ptr<!void>, cir.ptr <!cir.ptr<!void>>
+// CIR:    [[PTR:%.*]] = cir.load [[PTR_ALLOC]] : cir.ptr <!cir.ptr<!void>>, !cir.ptr<!void>
+// CIR:    cir.prefetch([[PTR]] : !cir.ptr<!void>, true, 1, true)
```

Comment:
My suggestion is that the final version looks something like:
`cir.prefetch([[PTR]] : !cir.ptr<!void>) locality(1) write`

Reply:
`cir.prefetch` was updated as you suggest